src/components/images/draw.js

```diff
@@ -84,31 +84,37 @@ module.exports = function draw(gd) {
             var img = new Image();
             this.img = img;
 
-            // If not set, a `tainted canvas` error is thrown
-            img.setAttribute('crossOrigin', 'anonymous');
-            img.onerror = errorHandler;
-            img.onload = function() {
-                var canvas = document.createElement('canvas');
-                canvas.width = this.width;
-                canvas.height = this.height;
+            if(d.source && d.source.slice(0, 5) === 'data:') {
```

Comment:
`d.source` should always be a string at this stage, so you could use:

```js
if(d.source.indexOf('data:') === 0) {
  // ...
}
```

instead.

Reply:
👍

Reply:
`d.source` can be large though, isn't `slice(0, 5)` going to be a lot faster than `indexOf`?

Reply:
That's a good point @alexcjohnson. @etpinard, anything you don't like about `slice` here?

Reply:
It's not obvious to me why `slice` would be faster than `indexOf() === 0`. Has anyone tested that out on jsperf before?

Reply:
My thinking was that If the string starts with `'data:'`, then `indexOf()` would terminate right away. But if not, `indexOf()` would search the full string and not bail out until it reaches the end, whereas slice never needs to look past the first 5 characters.  If the string doesn't start with `data:`, then I guess it should be a URL which hopefully isn't terribly long. So it might not matter either way.

Reply:
I guess in as far as the two options we care about are a short string starting `http`
and a long string starting `data:`, `indexOf` should be fine. The case to worry about would be a long string that doesn't start `data:`... but we don't expect that and wouldn't care if it's slow.

jsperf shows a win for `indexOf` https://jsperf.com/slice-vs-indexof-acj but I don't trust it, there's no way it could actually do `indexOf` in 2 clock cycles, compiler must have turned it into a constant. This isn't happening in a loop so either option is fast enough to be not worth worrying about.